### DISCUSSION TITLE
Realtime with multiple subs wont work

### DISCUSSION THREAD

**Discussion by Schrolli91** [appwrite/appwrite#7089] [Q&A] [answered]
I'am a bit curious how the realtime API will work.

Short example - if i have one sub, all is working fine and i will get my print on document changes.
```dart
if (_rt1 == null) {
  _rt1 = ref.watch(appwriteRealtimeProvider).subscribe(['documents']);
  _rt1!.stream.listen((event) {
    print('rt1');
  });
}
```

But if i will do the following and add a sub to the account endpoint, no one of my listen methods will be called anymore. In chrome DevTools i will see the WS connection and if i change a document, i will also get the packet trough the WS.
So the connection is still alive but no one of my callbacks are triggerd if is habe more than one...
```dart
if (_rt1 == null) {
  _rt1 = ref.watch(appwriteRealtimeProvider).subscribe(['account']);
  _rt1!.stream.listen((event) {
    print('rt1');
  });
}

await Future.delayed(Duration(seconds: 1));

if (_rt2 == null) {
  _rt2 = ref.watch(appwriteRealtimeProvider).subscribe(['account']);
  _rt2!.stream.listen((event) {
    print('rt2');
  });
}
```

What's wrong with my code, any ideas?

---
```
Restarted application in 2.142ms.
subscription: wss://####/v1/realtime?project=###-app&channels%5B%5D=account
subscription: wss://####/v1/realtime?project=###-app&channels%5B%5D=account&channels%5B%5D=documents
```

Docs are easy to understand, when it comes to ONE single connection. But to me, its not clear what I have to do, if I need a lot of subscriptions and how to manage them (central or decentral?/Separate Realtime() instances per subscription/listening callback?).

**Comment by stnguyen90** [comment, marked as answer]
This is a known issue with the flutter sdk: https://github.com/appwrite/sdk-for-flutter/issues/156

**Reply by Schrolli91** [reply]
So if i understand it correct - the way to go is:

- use a separate realtime instance for each sub
- NEVER resub on a realtime instance
- If i have to resub i should switch the complete rt instance with a new one?
